assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -10086,19 +10096,17 @@ private void putSubmissionInfoIntoState(SessionState state, String assignmentId,
                 state.setAttribute(ATTACHMENTS, v);
                 state.setAttribute(GRADE_SUBMISSION_FEEDBACK_ATTACHMENT, v);
                 String grade = assignmentService.getGradeDisplay(s.getGrade(), a.getTypeOfGrade(), a.getScaleFactor());
+                state.setAttribute(GRADE_SUBMISSION_GRADE, grade);
 
                 // populate grade overrides if they exist
-                for (AssignmentSubmissionSubmitter submitter : s.getSubmitters()) {
-                    String gradeOverride = assignmentService.getGradeDisplay(assignmentService.getGradeForSubmitter(s, submitter.getSubmitter()), a.getTypeOfGrade(), a.getScaleFactor());
-                    if (!StringUtils.equals(grade, gradeOverride)) {
-                        if (a.getIsGroup()) {
-                            state.setAttribute(GRADE_SUBMISSION_GRADE + "_" + submitter.getSubmitter(), gradeOverride);
-                        } else {
-                            grade = gradeOverride;
-                        }
-                    }
+                if (a.getIsGroup()) {
+	                for (AssignmentSubmissionSubmitter submitter : s.getSubmitters()) {
+	                    String gradeOverride = assignmentService.getGradeForSubmitter(s, submitter.getSubmitter());
+	                    if (!StringUtils.equals(grade, gradeOverride)) {
+	                        state.setAttribute(GRADE_SUBMISSION_GRADE + "_" + submitter.getSubmitter(), gradeOverride);
+	                    }
+	                }
                 }
-                state.setAttribute(GRADE_SUBMISSION_GRADE, grade);
```

Comment:
On this line, now we are setting on GRADE_SUBMISSION_GRADE the grade translated because we call the method
"String grade = assignmentService.getGradeDisplay(s.getGrade(), a.getTypeOfGrade(), a.getScaleFactor());"
on line 10088

Reply:
After that, when we build the context "build_instructor_grade_submission_context" on line 3397 we are calling
context.put("value_grade", assignmentService.getGradeDisplay((String) state.getAttribute(GRADE_SUBMISSION_GRADE), gradeType, assignment.isPresent() ? assignment.get().getScaleFactor() : null));

and the first param is (String) state.getAttribute(GRADE_SUBMISSION_GRADE) which is translated on different language.
This is wrong. We need always to call getGradeDisplay with the first param on English language.

Reply:
can we put some logic in getGradeDisplay to deal with internationalization?

Reply:
@ern please, take a look at: https://github.com/sakaiproject/sakai/pull/6583/